### PR TITLE
ntHits and ntCard integration

### PR DESCRIPTION
**Pull request by jowong4**
This PR contains the initial steps to building a repetitive Bloom filter using kmers determined by ntHits with a repeat threshold cutoff calculated using the output of ntCard.

### CODE REVIEW

File: data/Makefile [hunk 1]
```diff
@@ -641,6 +641,15 @@ minsize=2000
 %.histo: %.hist
 	sed -n 's/^f//p' $< | tr '\t' ' ' >$@
 
+# Count k-mers using ntCard.
+%_k$(k).histogram.txt: %.fq.gz
+	$(time) ntcard -t $(t) -k $(k) -o $@ $<
```
Comment (lcoombe) [inline data/Makefile]: This is picky, but to keep it consistent -- we generally use `$k` for single-letter variables, and only the parentheses with multi-letter variables (ex. `$(time)`)

File: data/Makefile [hunk 2]
```diff
@@ -641,6 +641,15 @@ minsize=2000
...
+
+# Generate repetitive k-mers using ntHits.
+%_k$(k).rep: %.fq.gz %_k$(k).histogram.txt
+	$(eval repeat_threshold := $(shell $(python) $(bin)/physlr find-ntcard-mode $*_k$(k).histogram.txt ))
+	$(time) nthits -t $(t) -k $(k) -c $(repeat_threshold) -p $* $<;
```
Comment (lcoombe) [inline data/Makefile]:
```suggestion
	$(time) nthits -t $(t) -k $(k) -c $(repeat_threshold) -p $* $<
```

File: data/Makefile [hunk 3]
```diff
@@ -648,6 +648,15 @@ minsize=2000
 %.histo: %.hist
 	sed -n 's/^f//p' $< | tr '\t' ' ' >$@
 
+# Count k-mers using ntCard.
+%_k$(k).histogram.txt: %.fq.gz
```
Comment (lcoombe) [inline data/Makefile]:
```suggestion
%_k$k.histogram.txt: %.fq.gz
```

File: data/Makefile [hunk 4]
```diff
@@ -648,6 +648,15 @@ minsize=2000
...
+	$(time) ntcard -t $t -k $k -o $@ $<
+
+# Generate repetitive k-mers using ntHits.
+%_k$(k).rep: %.fq.gz %_k$(k).histogram.txt
```
Comment (lcoombe) [inline data/Makefile]:
```suggestion
%_k$k.rep: %.fq.gz %_k$k.histogram.txt
```

File: data/Makefile [hunk 5]
```diff
@@ -648,6 +648,18 @@ minsize=2000
...
 	sed -n 's/^f//p' $< | tr '\t' ' ' >$@
 
+# Count k-mers using ntCard.
+%_k$k.histogram.txt: %.fq.gz
+	$(time) ntcard -t $t -k $k -o $@ $<
+
+# Generate repetitive k-mers using ntHits.
+%_k$k.rep: %.fq.gz %_k$k.histogram.txt
+	$(eval mode := $(shell $(python) $(bin)/physlr find-ntcard-mode $*_k$k.histogram.txt ))
+	>&2 echo "The mode k-mer coverage is $(mode)."
```
Comment (lcoombe) [inline data/Makefile]: We'll want `@echo` here so that it doesn't print out that line before it executes it. Do we need the `>&2`?

Comment (jowong4) [reply data/Makefile]: For the second part I'm not sure. Im ok with piping it to stdout by default, but I figured some people might only use `2> x.log`

Comment (lcoombe) [reply data/Makefile]:
```suggestion
	@>&2 echo "The mode k-mer coverage is $(mode)."
```
if you want to keep the stderr pipe

**Comment by jowong4** [comment]
> Looks good to me! So now we have those runs, and we'll integrate into the default pipeline in future pull requests?

That's the idea. Eventually  `make xyz` will call those new rules that I added

**Pull request merged**